axlearn/cloud/gcp/jobset_utils.py

```diff
@@ -717,80 +717,183 @@ def __call__(self) -> Sequence[Nested[Any]]:
+        env_vars: dict[str, str] = {}
+        env_vars["DISTRIBUTED_COORDINATOR"] = f"{cfg.name}-job-0-0.{cfg.name}:8080"
+        env_vars["NUM_PROCESSES"] = f"{cfg.accelerator.num_replicas}"
+
+        # List of XLA flags across all A3 and A4 instances
+        global_gpu_xla_flags = [
+            "--xla_gpu_enable_latency_hiding_scheduler=true",
+            "--xla_gpu_enable_triton_gemm=false",
+            "--xla_gpu_enable_pipelined_all_gather=true",
+            "--xla_gpu_enable_pipelined_reduce_scatter=true",
+            "--xla_gpu_enable_pipelined_all_reduce=true",
+            "--xla_gpu_enable_while_loop_double_buffering=true",
+            "--xla_gpu_enable_all_gather_combine_by_dim=false",
+            "--xla_gpu_enable_reduce_scatter_combine_by_dim=false",
+            "--xla_disable_hlo_passes=rematerialization",
+            " ", # Leave a trailing space for A3 / A4-specific XLA flags to be added
+        ]
+        env_vars["XLA_FLAGS"] = " ".join(global_gpu_xla_flags)
+
+        return dict(
+            name=cfg.name,
+            image=self._bundler.id(cfg.name),
+            ports=[
+                dict(containerPort=8080),  # Port for MXLA coordinator.
+            ],
+            securityContext=dict(privileged=True),
+            # TODO(markblee): Improve SIGTERM behavior for command.
+            resources=dict(limits={"nvidia.com/gpu": "8"}),
+            env=env_vars,
+            volumeMounts=volume_mounts,
+        )
+    
+    def _build_volumes(self) -> Nested[Any]:
+        """Builds a config for volumes."""
+        volumes = [
             {
-                "name": "nvidia-install-dir-host",
-                "mountPath": "/usr/local/nvidia/lib64",
+                "name": "shared-memory",
+                "emptyDir": {"medium": "Memory"},
             },
             {
-                "name": "tcpx-socket",
-                "mountPath": "/run/tcpx",
+                "name": "nvidia-install-dir-host",
+                "hostPath": {"path": "/home/kubernetes/bin/nvidia/lib64"},
             },
         ]
 
-        command = [
-            "bash",
-            "-c",
-            'set -x; /tcpgpudmarxd/build/app/tcpgpudmarxd --gpu_nic_preset a3vm  \
-                --gpu_shmem_type fd --uds_path /run/tcpx \
-                --setup_param "--verbose 128 2 0" & \n\
-            while [ ! -f /run/tcpx/terminated ]; do sleep 10; done;',
-        ]
+        return volumes
+    
+    def _build_pod(self) -> Nested[Any]:
+        """Builds a config for a single Pod, which is a set of containers.
+
+        https://kubernetes.io/docs/concepts/workloads/pods
+
+        Returns:
+            A nested dict corresponding to a k8s Pod template, including the pod metadata and spec.
+        """
+        cfg: GPUReplicatedJob.Config = self.config
+        volumes = self._build_volumes()
+        annotations = {
+            "kubectl.kubernetes.io/default-container": cfg.name,
```

Comment:
I understand this was also there in the original code. But curious if you know what this annotation is for?

https://kubernetes.io/docs/reference/labels-annotations-taints/#kubectl-kubernetes-io-default-container
Does it have any other implication besides annotating?

Reply:
It's helpful when you run things like `kubectl logs` and there are sidecar containers. We can probably get rid of it now that sidecar containers are part of initContainers.